source/extensions/filters/network/mongo_proxy/bson_impl.cc

```diff
@@ -101,7 +101,7 @@ std::string BufferHelper::removeString(Buffer::Instance& data) {
   }
 
   char* start = reinterpret_cast<char*>(data.linearize(length));
-  std::string ret(start);
+  std::string ret(start, strlen(start));
```

Comment:
this does not look right. I think maybe this should this be
```
std::string(ret(start, length);
```

But in any case, adding `strlen(start)` to as a constructor arg cannot help; that would be the default behavior anyway.

Reply:
I'm not sure about `length` being the right answer either, FWIW.

Reply:
Separately handled if `start` is not cstring.